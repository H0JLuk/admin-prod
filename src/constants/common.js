export const showCount = { formatter: ({ count, maxLength }) => `Осталось символов ${maxLength - count}` };

/* '^(https?:\\/\\/)?'+ // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
    '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
    '(\\:\\d+)?(\\/[-a-z\\d%_.~+{}]*)*'+ // port and path
    '(\\?[;&a-z\\d%_.~+=-{}]*)?'+ // query string
    '(\\#[-a-z\\d_]*)?$','i'); // fragment locator */
export const URL_REGEXP = /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(:\d+)?(\/[-A-Za-z\d%_.~+#]*)*(\?[;&a-z\d%_.~+=-{}]*)?(#[-a-z\d_]*)?$/;
