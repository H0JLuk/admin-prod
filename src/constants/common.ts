import { TextAreaProps } from 'antd/lib/input';

export const showCount: TextAreaProps['showCount'] = {
    formatter: ({ count, maxLength }) => `Осталось символов ${maxLength! - count}`,
};

export const DEFAULT_SLEEP_TIME = 500; // in milliseconds

export enum DIRECTION {
    ASC = 'ASC',
    DESC = 'DESC',
}
