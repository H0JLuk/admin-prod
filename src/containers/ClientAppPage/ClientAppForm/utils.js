import { notification } from 'antd';
import { addSettings, updateSettingsList } from '../../../api/services/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';

export const showNotify = (message, isError) => {
    const config = {
        duration: 5,
        placement: 'bottomRight',
    };

    if (isError) {
        notification.error({
            ...config,
            message,
        });
    } else {
        notification.success({
            ...config,
            message,
        });
    }

};

export async function createOrUpdateKey (changedParams) {
    const updateSettings = [];
    const addSettingsArr = [];

    changedParams.forEach(({ type, ...params }) => {
        type === SETTINGS_TYPES.EDIT
            ? updateSettings.push(params)
            : addSettingsArr.push(params);
    });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export const checkExistDesignSettings = (settings) => {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
};

export const getImage = (imageName) => {
    return `${process.env.PUBLIC_URL}/images/clientAppDesign/${imageName}.png`;
};


