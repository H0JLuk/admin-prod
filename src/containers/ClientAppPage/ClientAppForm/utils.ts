import { customNotifications } from '@utils/notifications';
import { ArgsProps } from 'antd/lib/notification';
import { addSettings, updateSettingsList } from '@apiServices/settingsService';
import { SETTINGS_TYPES } from './ClientAppFormConstants';
import { SettingDto } from '@types';
import ROLES from '@constants/roles';

export interface IChangedParam extends Pick<SettingDto, 'clientAppCode' | 'key' | 'value' | 'userRole'> {
    type: SETTINGS_TYPES;
}

export const showNotify = (message: React.ReactNode, isError?: boolean) => {
    const config: ArgsProps = {
        duration: 5,
        message,
    };

    if (isError) {
        customNotifications.error(config);
    } else {
        customNotifications.success(config);
    }

};

type UpdateAndAddSettings = {
    updateSettings: SettingDto[];
    addSettingsArr: SettingDto[];
};

export async function createOrUpdateKey(changedParams: IChangedParam[]) {
    const { updateSettings, addSettingsArr } = changedParams.reduce<UpdateAndAddSettings>((result, { type, ...params }) => {
        const key = type === SETTINGS_TYPES.EDIT ? 'updateSettings' : 'addSettingsArr';

        if (params.key === 'referralTokenLifetime') {
            delete params.clientAppCode;
            params.key = 'token_lifetime';
            params.userRole = ROLES.REFERAL_LINK;
        }

        return {
            ...result,
            [key]: [...result[key], params],
        };
    }, { updateSettings: [], addSettingsArr: [] });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}
