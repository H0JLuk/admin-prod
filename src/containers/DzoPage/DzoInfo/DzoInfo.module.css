.dzoWrapper {
    width: 100%;
    height: 100%;
}

.container {
    position: relative;
    display: flex;
    flex-direction: column;
    padding: 0px 60px 0px 15px;
}

.infoWrapper {
    padding: 30px;
    background-color: #f8f8f8;
    margin-bottom: 25px;
}

.dzoInfo {
    max-width: 800px;
}

.title {
    margin-bottom: 6px;
}

.dzoTitle {
    font-weight: 600;
    font-size: 20px;
    line-height: 28px;
    letter-spacing: -.38px;
    margin-top: 24px;
    margin-bottom: 16px;
}

.text {
    font-weight: 600;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
}

.footer {
    margin-bottom: 25px;
}

.bannersRow {
    padding: 26px 0 46px;
}

.infoTitle {
    margin-bottom: 6px;
    font-size: 16px;
    font-weight: bold;
    line-height: 26px;
    letter-spacing: -0.3px;
}

.dzoInfoRow {
    margin-bottom: 26px;
}