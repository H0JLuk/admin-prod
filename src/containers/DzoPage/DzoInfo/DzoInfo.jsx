import React from 'react';
import { generatePath, useHistory, useLocation } from 'react-router-dom';
import { Col, Row, Space, Button } from 'antd';
import Header from '../../../components/Header/Header';
import { DZO_PAGES } from '../../../constants/route';
import { deleteDzo } from '../../../api/services/dzoService';
import { confirmModal, errorModal, successModal } from '../../../utils/utils';

import styles from './DzoInfo.module.css';

const DZO_NAME = 'Название';
const DZO_CODE = 'Код';
const DZO_DESCRIPTION = 'Описание';
const DZO_APP_TYPE = 'Тип приложения';
const DZO_QR_URL = 'Ссылка для QR-кода';
const EDIT = 'Редактировать';
const DELETE = 'Удалить';
const OK_TEXT = 'Хорошо';
const DELETE_CONFIRMATION_MODAL_TITLE = 'Вы действительно хотите удалить ДЗО';
const ERROR_DELETE_DZO = 'Ошибка удаления ДЗО';

const DZO_INFO_TEMPLATE_DATA = [
    [
        { key: 'dzoName', label: DZO_NAME },
        { key: 'dzoCode', label: DZO_CODE },
    ],
    [
        { key: 'description', label: DZO_DESCRIPTION },
    ],
];

const DZO_INFO_APPS_TEMPLATE = [
    { key: 'applicationType', label: DZO_APP_TYPE },
    { key: 'applicationUrl', label: DZO_QR_URL, type: 'url' },
];

const DzoInfo = ({ matchPath }) => {
    const history = useHistory();
    const { state: stateFromLocation } = useLocation();
    const { dzoData } = stateFromLocation || {};
    const redirectToDzoList = () => history.push(matchPath);
    if (!dzoData) {
        history.push(matchPath);
        return null;
    }

    const { dzoId, applicationList } = dzoData;

    const onEdit = () => {
        history.push(generatePath(`${ matchPath }${ DZO_PAGES.DZO_EDIT }`, { dzoId }), { dzoData });
    };

    const onDelete = async () => {

        try {
            await deleteDzo(dzoId);

            // TODO: maybe later change this to notification
            successModal({
                onOk: redirectToDzoList,
                title: (
                    <span>
                        ДЗО <span className={ styles.text }>{ dzoData.dzoName }</span> успешно удалено
                    </span>
                ),
                okText: OK_TEXT,
            });
        } catch (e) {
            const { message } = e;

            errorModal({
                title: (
                    <span>
                        { ERROR_DELETE_DZO } <span className={ styles.text }>{ message }</span>
                    </span>
                ),
            });
            console.warn(e);
        }

    };

    const onDeleteClick = () => {
        confirmModal({
            title: (
                    <span>
                        { DELETE_CONFIRMATION_MODAL_TITLE } <span className={ styles.text }>{ `${dzoData.dzoName}?` }</span>
                    </span>
                ),
            okText: DELETE,
            onOk: onDelete,
            okButtonProps: { danger: true },
        });
    };

    return (
        <div className={ styles.dzoWrapper }>
            <Header />
            <div className={ styles.container }>
                    <div className={ styles.dzoTitle }>
                        { dzoData.dzoName }
                    </div>
                <div className={ styles.infoWrapper }>
                    <div className={ styles.dzoInfo }>
                        { DZO_INFO_TEMPLATE_DATA.map((row, index) => (
                            <Row gutter={ [24, 16] } key={ index }>
                                { row.map(({ key, label }) => (
                                    <DzoInfoBlock
                                        key={ key }
                                        label={ label }
                                        value={ dzoData[key] }
                                        colSpan={ 24 / row.length }
                                    />
                                )) }
                            </Row>
                        )) }
                        { applicationList.map(dzoApp => (
                            <Row gutter={ [24, 16] } key={ dzoApp.applicationId }>
                                { DZO_INFO_APPS_TEMPLATE.map(({ key, label, type }) => (
                                    <DzoInfoBlock
                                        key={ key }
                                        label={ label }
                                        value={ dzoApp[key] }
                                        type={ type }
                                        colSpan={ 12 }
                                    />
                                )) }
                            </Row>
                        )) }
                    </div>
                </div>
                <div className={ styles.footer }>
                    <Space>
                        <Button
                            type="primary"
                            onClick={ onEdit }
                        >
                            { EDIT }
                        </Button>
                        <Button
                            type="primary"
                            danger
                            onClick={ onDeleteClick }
                        >
                            { DELETE }
                        </Button>
                    </Space>
                </div>
            </div>
        </div>
    );
};

export default DzoInfo;

function DzoInfoBlock({ label, value, type, colSpan }) {
    return (
        <Col span={ colSpan }>
            <div className={ styles.title }>{label}</div>
            <span className={ styles.text }>
                {type === 'url' ? decodeURI(value) : value}
            </span>
        </Col>
    );
}
