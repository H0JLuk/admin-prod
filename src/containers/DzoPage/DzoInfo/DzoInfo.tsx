import React from 'react';
import { generatePath, useHistory, useLocation } from 'react-router-dom';
import { Col, Row, Space, Button } from 'antd';
import Header from '../../../components/Header/Header';
import { DZO_PAGES } from '../../../constants/route';
import { deleteDzo } from '../../../api/services/dzoService';
import { confirmModal, errorModal, successModal } from '../../../utils/utils';
import { ImageBlock } from '../../PromoCampaignPage/PromoCampaign/PromoCampaignInfo/Steps/Templates/TemplateBlocks';
import {
    DELETE_BUTTON_LABEL,
    DELETE_CONFIRMATION_MODAL_TITLE,
    ERROR_DELETE_DZO, OK_TEXT,
    DZO_INFO_TEMPLATE_DATA,
    DZO_BANNERS_TEMPLATE,
    BANNER_IS_EMPTY,
    DZO_INFO_APPS_TEMPLATE,
    EDIT,
    IMainDataRow,
    IDzoBannersRow,
    IDzoInfoRow,
} from '../dzoConstants';
import { dzoBannerTypes, IDzoItem } from '@types';

import styles from './DzoInfo.module.css';

interface IDzoInfoProps {
    matchPath: string;
}

interface ILocationState {
    dzoData: IDzoItem;
}

type INormalizedBannerList = {
    [BannerType in keyof typeof dzoBannerTypes]?: string;
};

interface IDzoInfoBlock {
    label: string;
    value: string;
    type?: string ;
    colSpan: number;
}

const DzoInfo = ({ matchPath }: IDzoInfoProps) => {
    const history = useHistory();
    const { state: stateFromLocation } = useLocation();
    const { dzoData } = (stateFromLocation || {}) as ILocationState;
    const redirectToDzoList = () => history.push(matchPath);

    if (!dzoData) {
        history.push(matchPath);
        return null;
    }

    const { dzoId, applicationList, dzoBannerList } = dzoData;
    const normalizedBannerList: INormalizedBannerList = dzoBannerList.reduce((acc, { type, url }) => ({
        ...acc,
        [type]: url,
    }), {});

    const onEdit = () => {
        history.push(generatePath(`${ matchPath }${ DZO_PAGES.DZO_EDIT }`, { dzoId }), { dzoData });
    };

    const onDelete = async () => {
        try {
            await deleteDzo(dzoId);

            // TODO: maybe later change this to notification
            successModal({
                onOk: redirectToDzoList,
                title: (
                    <span>
                        ДЗО <span className={styles.text}>{dzoData.dzoName}</span> успешно удалено
                    </span>
                ),
                okText: OK_TEXT,
            });
        } catch (e) {
            const { message } = e;

            errorModal({
                title: (
                    <span>
                        {ERROR_DELETE_DZO} <span className={styles.text}>{message}</span>
                    </span>
                ),
            });
            console.warn(e);
        }
    };

    const onDeleteClick = () => {
        confirmModal({
            title: (
                <span>
                    {DELETE_CONFIRMATION_MODAL_TITLE} <span className={styles.text}>{`${dzoData.dzoName}?`}</span>
                </span>
            ),
            okText: DELETE_BUTTON_LABEL,
            onOk: onDelete,
            okButtonProps: { danger: true },
        });
    };

    return (
        <div className={styles.dzoWrapper}>
            <Header />
            <div className={styles.container}>
                <div className={styles.dzoTitle}>
                    {dzoData.dzoName}
                </div>
                <div className={styles.infoWrapper}>
                    <div className={styles.dzoInfo}>
                        {DZO_INFO_TEMPLATE_DATA.map((row, index) => (
                            <Row gutter={[24, 16]} key={index} className={styles.dzoInfoRow}>
                                {row.map(({ key, label }: IMainDataRow) => (
                                    <DzoInfoBlock
                                        key={key}
                                        label={label}
                                        value={dzoData[key] || ''}
                                        colSpan={24 / row.length}
                                    />
                                ))}
                            </Row>
                        ))}
                        <Row className={styles.bannersRow} gutter={24}>
                            {DZO_BANNERS_TEMPLATE.map(({ type, label }: IDzoBannersRow) =>
                                <Col span={8} key={type}>
                                    {normalizedBannerList[type] ? (
                                        <ImageBlock
                                            type="logo"
                                            label={label}
                                            src={normalizedBannerList[type]}
                                        />
                                    ) : (
                                        <>
                                            <div className={styles.infoTitle}>{label}</div>
                                            <div>
                                                <i>{BANNER_IS_EMPTY}</i>
                                            </div>
                                        </>
                                    )}
                                </Col>
                            )}
                        </Row>
                        <Row gutter={[24, 16]}>
                            {applicationList.map(dzoApp =>
                                DZO_INFO_APPS_TEMPLATE.map(({ key, label, type }: IDzoInfoRow) => (
                                    <DzoInfoBlock
                                        key={key}
                                        label={label}
                                        value={dzoApp[key]}
                                        type={type}
                                        colSpan={12}
                                    />
                                ))
                            )}
                        </Row>
                    </div>
                </div>
                <div className={styles.footer}>
                    <Space>
                        <Button
                            type="primary"
                            onClick={onEdit}
                        >
                            {EDIT}
                        </Button>
                        <Button
                            type="primary"
                            danger
                            onClick={onDeleteClick}
                        >
                            {DELETE_BUTTON_LABEL}
                        </Button>
                    </Space>
                </div>
            </div>
        </div>
    );
};

export default DzoInfo;

function DzoInfoBlock({ label, value, type, colSpan }: IDzoInfoBlock) {
    const decodedUrl = type === 'url' ? decodeURI(value) : value;

    return (
        <Col span={colSpan}>
            <div className={styles.title}>{label}</div>
            <span className={styles.text}>
                {value ? decodedUrl : (<i>{`${ label } отсутствует`}</i>)}
            </span>
        </Col>
    );
}
