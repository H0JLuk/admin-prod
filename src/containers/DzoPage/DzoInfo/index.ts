import DzoInfo from './DzoInfo';

export default DzoInfo;
