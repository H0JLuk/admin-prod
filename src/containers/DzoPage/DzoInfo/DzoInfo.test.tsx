import React from 'react';
import { generatePath, useLocation } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { confirmModal, errorModal, successModal } from '@utils/utils';
import { BUTTON_TEXT } from '@constants/common';
import {
    deleteDzo,
} from '../../../api/services/dzoService';
import DzoInfo from './DzoInfo';
import { dzoListTestData } from '../../../../__tests__/constants';

const props = {
    matchPath: 'matchPath',
};
const mockHistoryPush = jest.fn();

jest.mock('@utils/utils', () => ({
    confirmModal: jest.fn(),
    successModal: jest.fn(),
    errorModal: jest.fn(),
}));

jest.mock('../../../api/services/dzoService', () => ({
    deleteDzo: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn(),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
    generatePath: jest.fn(),
}));

const dzoId = dzoListTestData[0].dzoId;
const mockLocationState = {
    dzoCodes: ['doc_old_4', 'delivery_club_old_3', 'sberbox'],
    dzoData: dzoListTestData[0],
};

describe('<DzoInfo /> test', () => {
    beforeEach(() => {
        (useLocation as jest.Mock).mockImplementation(() => ({
            state: mockLocationState,
        }));
    });

    it('should match the snapshot', () => {
        const Component = render(<DzoInfo {...props} />);
        expect(Component.container).toMatchSnapshot();
    });

    it('should redirect without dzoData from location state', () => {
        (useLocation as jest.Mock).mockImplementation(() => ({ state: {} }));
        render(<DzoInfo {...props} />);
        expect(mockHistoryPush).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledWith(props.matchPath);
    });

    it('should edit dzo', () => {
        (generatePath as jest.Mock).mockReturnValue(props.matchPath);
        render(<DzoInfo {...props} />);

        userEvent.click(screen.getByText(BUTTON_TEXT.EDIT));
        const { dzoData } = mockLocationState;
        expect(mockHistoryPush).toBeCalledWith(props.matchPath, { dzoData });
    });

    it('should delete dzo after submit modal window', async () => {
        (deleteDzo as jest.Mock).mockResolvedValue(null);
        render(<DzoInfo {...props} />);

        userEvent.click(screen.getByText(BUTTON_TEXT.DELETE));
        expect(confirmModal).toHaveBeenCalledTimes(1);
        await (confirmModal as jest.Mock).mock.calls[0][0].onOk();
        expect(deleteDzo).toBeCalledWith(dzoId);

        await (successModal as jest.Mock).mock.calls[0][0].onOk();
        expect(mockHistoryPush).toBeCalledWith(props.matchPath);
    });

    it('should handle error at deleting dzo after submit modal window', async () => {
        const consoleWarn = jest.spyOn(console, 'warn').mockImplementation(() => '');
        (deleteDzo as jest.Mock).mockRejectedValue(new Error('mock error message'));
        render(<DzoInfo {...props} />);

        userEvent.click(screen.getByText(BUTTON_TEXT.DELETE));
        expect(confirmModal).toHaveBeenCalledTimes(1);
        await (confirmModal as jest.Mock).mock.calls[0][0].onOk();

        expect(deleteDzo).toBeCalledWith(dzoId);
        expect(consoleWarn).toBeCalledWith(new Error('mock error message'));
        expect(errorModal).toBeCalledTimes(1);
    });
});
