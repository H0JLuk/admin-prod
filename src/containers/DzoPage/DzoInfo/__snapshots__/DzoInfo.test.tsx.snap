// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DzoInfo /> test should match the snapshot 1`] = `
<div>
  <div
    class="dzoWrapper"
  >
    <div
      class="container"
    >
      <div
        class="backButton"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
         
         Назад
      </div>
      <div
        class="avatar"
      >
        <span
          aria-label="user"
          class="anticon anticon-user ant-dropdown-trigger dropdown"
          role="img"
          style="font-size: 22px; color: rgb(255, 255, 255); background-color: rgb(193, 200, 209); border-radius: 50px; width: 32px; height: 32px; padding: 3px;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="dzoTitle"
      >
        Беру!
      </div>
      <div
        class="infoWrapper"
      >
        <div
          class="dzoInfo"
        >
          <div
            class="ant-row dzoInfoRow"
            style="margin: -8px -12px -8px -12px;"
          >
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Название
              </div>
              <div
                class="text"
              >
                Беру!
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Код
              </div>
              <div
                class="text"
              >
                beru_code
              </div>
            </div>
          </div>
          <div
            class="ant-row dzoInfoRow"
            style="margin: -8px -12px -8px -12px;"
          >
            <div
              class="ant-col ant-col-24"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Описание
              </div>
              <div
                class="text"
              >
                Простой поиск и дешевая покупка любых товаров онлайн у проверенных продавцов
              </div>
            </div>
          </div>
          <div
            class="ant-row bannersRow"
            style="margin-left: -12px; margin-right: -12px;"
          >
            <div
              class="ant-col ant-col-8"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="title"
              >
                Логотип (цветной)
              </div>
              <div
                class="block"
              >
                <div
                  class="logo"
                >
                  <img
                    alt=""
                    class="imageLogo"
                    src="https://goo.gle/icon.png"
                  />
                </div>
                <div
                  class="footer"
                >
                  <div
                    class="fileName"
                  >
                    icon.png
                  </div>
                  <div
                    class="fileSize"
                  />
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-8"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="infoTitle"
              >
                Логотип (белый)
              </div>
              <div>
                <i>
                  Логотип отсутствует
                </i>
              </div>
            </div>
            <div
              class="ant-col ant-col-8"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="infoTitle"
              >
                Иконка
              </div>
              <div>
                <i>
                  Логотип отсутствует
                </i>
              </div>
            </div>
          </div>
          <div
            class="ant-row dzoInfoRow"
            style="margin: -8px -12px -8px -12px;"
          >
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Тип приложения
              </div>
              <div
                class="text"
              >
                IOS
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Ссылка для QR-кода
              </div>
              <div
                class="text"
              >
                https://apps.apple.com/ru/app/id1369890634
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Тип приложения
              </div>
              <div
                class="text"
              >
                ANDROID
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
              style="padding: 8px 12px 8px 12px;"
            >
              <div
                class="title"
              >
                Ссылка для QR-кода
              </div>
              <div
                class="text"
              >
                https://play.google.com/store/apps/details/id1369890634
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -12px; margin-right: -12px;"
          >
            <div
              class="ant-col ant-col-12"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="title"
              >
                Ссылка на видеоэкскурсию
              </div>
              <div
                class="text"
              >
                <i>
                  Ссылка на видеоэкскурсию отсутствует
                </i>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Редактировать
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-dangerous"
              type="button"
            >
              <span>
                Удалить
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
